### PR TITLE
deprecate the mutability of Gc.control record fields

### PR DESCRIPTION
**Pull request by damiendoligez**
Follow-up to https://github.com/ocaml/ocaml/pull/297#discussion_r46124492. I'll add a `Changes` entry.

### CODE REVIEW

File: stdlib/gc.mli [hunk 1]
```diff
@@ -84,17 +84,21 @@ type stat =
 
 type control =
   { mutable minor_heap_size : int;
+    [@ocaml.deprecated_mutable "Use Gc.{(get()) with minor_heap_size = ...}"]
```
Comment (alainfrisch) [inline stdlib/gc.mli]: I would personally recommend using `{(Gc.get()) with ...} `.  A local-open could lead to:
````ocaml
  let major = 42 in
  Gc.(set {(get ()) with major_heap_increment = major}) (* boom!  major is shadow by local open *)
````

**Review by alainfrisch — APPROVED** [review]
LGTM, except I would prefer documenting a syntax without the local-open.

File: stdlib/gc.mli (hunk 1 above)
Comment (damiendoligez) [reply stdlib/gc.mli]: Done.

**Comment by damiendoligez** [comment]
As far as I can tell, Appveyor failed because of a time-out. Since this is a low-risk change I'll merge rather than relaunch all the tests.

**Pull request merged**